### PR TITLE
refactor!: replace external vm interface and receipts with local ones

### PR DESCRIPTION
**Pull request by austinabell**
Removes need for https://github.com/near/nearcore/pull/4101 through just creating `SdkExternal` to satisfy the interface done by vm_logic before.

Built on #451 as it's a cleaner change to not have to deal with conflicts so leaving in draft for now

This exposes Receipts and the sub fields to be able to be used in tests

### CODE REVIEW

File: near-sdk/src/environment/mock/receipt.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#[derive(Clone, Debug)]
+pub struct Receipt {
+    pub receipt_indices: Vec<u64>,
+    pub receiver_id: String,
+    pub actions: Vec<VmAction>,
+}
+
+#[derive(Clone, Debug)]
+pub enum VmAction {
```
Comment (matklad) [inline near-sdk/src/environment/mock/receipt.rs:12]: Let's slap `#[non_exhaustive]` here

File: near-sdk/src/environment/mock/receipt.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
+use crate::{AccountId, Balance, Gas, PublicKey};
+// use near_vm_logic::mocks::mock_external::Action as VmAction;
```
Comment (matklad) [inline near-sdk/src/environment/mock/receipt.rs:2]: stary comment?

File: near-sdk/src/environment/mock/receipt.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    AddKeyWithFullAccess {
+        public_key: PublicKey,
+        nonce: u64,
+    },
+    AddKeyWithFunctionCall {
+        public_key: PublicKey,
+        nonce: u64,
+        allowance: Option<Balance>,
+        receiver_id: AccountId,
+        method_names: Vec<Vec<u8>>,
```
Comment (ChaoticTempest) [inline near-sdk/src/environment/mock/receipt.rs:43]: I see that `method_names` in `mock_external.rs` also has a `#[serde(with = "crate::serde_with::vec_bytes_as_str")]`. Would these vm actions need it too?

Comment (austinabell) [reply near-sdk/src/environment/mock/receipt.rs:43]: no, no need for receipts to be serializable (yet)

**Review by ChaoticTempest — APPROVED** [review]

File: near-sdk/src/environment/mock/receipt.rs [hunk 4]
```diff
@@ -0,0 +1,51 @@
+use crate::{AccountId, Balance, Gas, PublicKey};
+
+#[derive(Clone, Debug)]
+pub struct Receipt {
```
Comment (mikedotexe) [inline near-sdk/src/environment/mock/receipt.rs:4]: This will be quite helpful, thanks

**Review by mikedotexe — APPROVED** [review]

**Pull request merged**